File: Easy/7-Segment Scanner/7-Segment Scanner.cs
using System;

class Solution
{
    static void Main(string[] args)
    {
        string line1 = Console.ReadLine();
        string line2 = Console.ReadLine();
        string line3 = Console.ReadLine();

        for (int i = 0; i<line1.Length/3; i++)
        {   
            string num1 = line1.Substring(i*3,3); 
            string num2 = line2.Substring(i*3,3); 
            string num3 = line3.Substring(i*3,3); 

            if (num1.Trim() == string.Empty)
                if (num2[1] == '_')
                    Console.Write(4);
                else 
                    Console.Write(1);

            else if (num2[0] == ' ')
                if (num2[1] == ' ')
                    Console.Write(7);
                else if (num3[0] == ' ')
                    Console.Write(3);
                else 
                    Console.Write(2);

            else if (num3[0] == ' ')
                if (num2[2] == ' ')
                    Console.Write(5);
                else
                    Console.Write(9);

            else if (num2[1] == ' ')
                Console.Write(0);
            else if (num2[2] == ' ')
                Console.Write(6);
    
            else
                Console.Write(8);
        }
    }
}

File: Easy/7-Segment Scanner/7-Segment Scanner.py
line_1 = input()
line_2 = input()
line_3 = input()

for i in range(int(len(line_1)/3)):
    l1 = line_1[i*3:(i*3)+3]
    l2 = line_2[i*3:(i*3)+3]
    l3 = line_3[i*3:(i*3)+3]

    if not l1.strip():
        if l2[1] == '_':
            print(4, end="")
        else:
            print(1, end="")
    elif l2[0] == ' ':
        if l2[1] == ' ':
            print(7, end="")
        elif l3[0] == ' ':
            print(3, end="")
        else:
            print(2, end="")

    elif l3[0] == ' ':
        if l2[2] == ' ':
            print(5, end="")
        else:
            print(9, end="")

    elif l2[1] == ' ':
        print(0, end="")
    elif l2[2] == ' ':
        print(6, end="")
    
    else:
        print(8,end="")

